ArmoniK.Extension.CSharp.Client.DllHelper/DynamicLibraryExt.cs

```diff
@@ -0,0 +1,121 @@
+// This file is part of the ArmoniK project
+// 
+// Copyright (C) ANEO, 2021-2024. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using ArmoniK.Api.gRPC.V1;
+using ArmoniK.Extension.CSharp.Client.Common.Domain.Task;
+
+namespace ArmoniK.Extension.CSharp.Client.DllHelper;
+
+/// <summary>
+///   Represents the configuration of an application, including its name, version, namespace, service, and engine type.
+/// </summary>
+public record DynamicLibrary
+{
+  /// <summary>
+  ///   FileName of the Dll.
+  /// </summary>
+  public string Name { get; init; }
+
+  /// <summary>
+  ///   FileName of the Dll.
+  /// </summary>
+  public string DllFileName { get; init; }
+
+  /// <summary>
+  ///   Version of the application.
+  /// </summary>
+  public string PathToFile { get; init; }
+
+  /// <summary>
+  ///   Namespace of the application.
+  /// </summary>
+  public string Namespace { get; init; }
+
+  /// <summary>
+  ///   Service name of the application.
+  /// </summary>
+  public string Service { get; init; }
+
+  /// <summary>
+  ///   Version of the application.
+  /// </summary>
+  public string Version { get; init; }
+
+  /// <summary>
+  ///   Library Blob Identifier.
+  /// </summary>
+  public string LibraryBlobId { get; set; }
+
+  public override string ToString()
+    => $"{Name}-{Namespace}-{Version}";
```

Comment:
A single DLL can have multiple namespaces in it. At least historically.
A DLL is a name and a version
Then inside the library the function is defined by a service in a namespace

Reply:
okay, I see better now. So I think I will change it and we will use mostly Name and Version to define a Dll